src/topology/builder.rs

```diff
@@ -75,7 +77,16 @@ pub async fn build_pieces(
         };
 
         let (output, control) = Fanout::new();
-        let pump = rx.map(Ok).forward(output).map_ok(|_| TaskOutput::Source);
+        let pump = rx
+            .map(move |event| {
+                decoding
+                    .clone()
```

Comment:
Is it possible to avoid cloning `decoding` for each event? I see `decoder.decode` isn't fully implemented yet, but it seems like it could take a `&self` rather than `self` to avoid the clone here.